addon/mixins/scoped-query-cache.js

```diff
@@ -73,5 +76,21 @@ export default Ember.Mixin.create({
 
       return results;
     });
+  },
+
+  scheduleCacheDecay(delay = CACHE_DECAY_TIMEOUT, callback) {
+    this.decayId = setTimeout(() => {
+      this.cache = this._createCache();
+
+      callback && typeof callback === 'function' && callback();
```

Comment:
Feel free to provide a specific example of what you'd like to see in code. This will help us get on the same page, and move this towards the final API design.